src/database.ts

```diff
@@ -599,38 +729,51 @@ class Database extends ServiceObject {
    *   const apiResponse = data[0];
    * });
    */
-  get(options?: GetDatabaseOptions): Promise<DatabaseResponse>;
-  get(options: GetDatabaseOptions, callback: DatabaseCallback): void;
-  get(callback: DatabaseCallback): void;
   get(
-    optionsOrCallback?: GetDatabaseOptions | DatabaseCallback,
+    optionsOrCallback?: GetConfig | DatabaseCallback,
     cb?: DatabaseCallback
-  ): void | Promise<DatabaseResponse> {
+  ): void | Promise<
+    [database_admin_client.spanner.admin.database.v1.Database, r.Response]
+  > {
     const options =
       typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
     const callback =
-      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
     this.getMetadata((err, metadata) => {
       if (err) {
         if (options.autoCreate && (err as ApiError).code === 5) {
-          this.create(options, (err, database, operation) => {
-            if (err) {
-              callback!(err);
-              return;
+          this.create(
+            options,
+            (
+              err,
+              database: database_admin_client.spanner.admin.database.v1.Database,
+              operation: database_admin_client.longrunning.Operations
+            ) => {
+              if (err) {
+                callback!(err);
+                return;
+              }
+              operation
+                .on('error', callback)
+                .on('complete', (metadata: Metadata) => {
+                  this.metadata = metadata;
+                  callback(null, this, metadata);
+                });
             }
-            operation!.on('error', callback!).on('complete', metadata => {
-              this.metadata = metadata;
-              callback!(null, this, metadata);
-            });
-          });
+          );
           return;
         }
-        callback!(err);
+        callback(err);
         return;
       }
-      callback!(null, this, metadata);
+      callback(null, this, metadata);
     });
   }
+
+  getMetadata(): Promise<
+    [database_admin_client.spanner.admin.database.v1.Database, r.Response]
```

Comment:
rather use Interface here and below. And I do not think r.Response is returned here. [link](https://github.com/laljikanjareeya/nodejs-spanner/blob/master/proto/spanner_database_admin.d.ts#L185)
```suggestion
    [database_admin_client.spanner.admin.database.v1.IDatabase]
```

Reply:
Response is required as per document below

Reply:
You can remove `response` from the return signatures and from the documentation.

Reply:
Done!!